pkg/terra/relay.go

```diff
@@ -100,13 +102,15 @@ func (r *Relayer) Healthy() error {
 	return r.chainSet.Healthy()
 }
 
-func (r *Relayer) NewOCR2Provider(externalJobID uuid.UUID, s interface{}) (relaytypes.OCR2Provider, error) {
+// NewOCR2Provider creates a new OCR2ProviderCtx instance.
+func (r *Relayer) NewOCR2Provider(externalJobID uuid.UUID, s interface{}) (relaytypes.OCR2ProviderCtx, error) {
```

Comment:
If you don't mind me asking, why the `Ctx` suffix in `OCR2ProviderCtx`?

Reply:
To distinguish the old interface and the new interface. We agreed that this suffix will indicate that context is used there.

Reply:
Should we add `Ctx` suffix to all types using context?

If you removed the old interface altogether, meaning we currently don't have two (why would we, this is new code used only internally for now so we can change it and break it atm), the `Ctx` suffix makes no sense to me 🤷

Reply:
I think we were worried about the solana and terra import cycles being a problem, but perhaps that is not an issue?

Reply:
Currently we have dual interfaces, but after merging everything, we will remove the old interfaces (see https://github.com/smartcontractkit/chainlink-solana/pull/245). Renaming everything back to Ctx-less naming will be hard, because we will need to simultaneously change terra/solana and core packages. I do not see an issue if the new (and the only) service will keep Ctx suffix.